### PR TITLE
[RAPTOR-3220] don't log exception if args were not parsed

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Two fixes here:
1. Fix indentation for os._exit call in a SIGINT handler.
2. Showing error messages. Drum is used locally and in the env in the app.
Locally we want to show nice error message without stacktrace and in a server mode we want to see stacktrace.

Maybe we want to add a flag `--show-stacktrace` and use it in `start_server.sh` and `fit.sh`?

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/main.py [hunk 1]
```diff
@@ -64,8 +64,8 @@ def signal_handler(sig, frame):
                     url = "http://{}/shutdown/".format(runtime.options.address)
                     print("Sending shutdown to server: {}".format(url))
                     requests.post(url, timeout=2)
-                os.system("tput init")
-                os._exit(130)
+            os.system("tput init")
```
Comment (rvorobii) [inline custom_model_runner/datarobot_drum/drum/main.py:67]: :+1:

File: custom_model_runner/datarobot_drum/drum/runtime.py [hunk 2]
```diff
@@ -27,14 +29,20 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
...
 
-        if not self.options:
-            # exception occurred before args were parsed
-            return False  # propagate exception further
+        if self.options.show_stacktrace:
+            logger.error(exc_msg)
+        else:
+            if exc_type == DrumCommonException:
+                print(exc_value)
+            exit(1)
```
Comment (rvorobii) [inline custom_model_runner/datarobot_drum/drum/runtime.py:45]: why we exit here?
if user passes `--with-error-server`, if exception is not `DrumCommonException` - the server will not be started

IMO it's better to:
```
if run_mode == RunMode.SERVER or self.options.show_stacktrace:
    logger.error(exc_msg)
```
-- we check is drum is run in server mode or if user passes `--show-stacktrace`, and if so - log error

File: custom_model_runner/datarobot_drum/drum/runtime.py [hunk 3]
```diff
@@ -37,14 +39,17 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         exc_msg_lines = traceback.format_exception(exc_type, exc_value, exc_traceback)
         exc_msg = "".join(exc_msg_lines)
 
-        if self.options.show_stacktrace:
+        if (
+            self.options.show_stacktrace
+            or run_mode != RunMode.SERVER
```
Comment (rvorobii) [inline custom_model_runner/datarobot_drum/drum/runtime.py]:
```suggestion
            or (run_mode == RunMode.SERVER
```

File: custom_model_runner/datarobot_drum/drum/runtime.py [hunk 4]
```diff
@@ -37,14 +39,17 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
...
+            and self.options.with_error_server
```
Comment (rvorobii) [inline custom_model_runner/datarobot_drum/drum/runtime.py:45]:
```suggestion
            and self.options.with_error_server)
```

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**7.58** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request merged**